nayanaamahesh iP

nayanaamahesh opened the pull request:
### NayanaPro
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

 NayanaPro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
-  ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nayanaamahesh/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
- [ ] 
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Level 0: Rename, Greet, Exit

nayanaamahesh committed:
Level 1: Echo

nayanaamahesh committed:
Level 2: Add, List

nayanaamahesh committed:
Level 3: Mark as Done

nayanaamahesh committed:
Level 4: Todo Event Deadlines

nayanaamahesh committed:
Level 5: Handle Errors & Level 6: Delete

nayanaamahesh committed:
Week 2 (Draft 1)

nayanaamahesh committed:
Added Testcases Draft 1

nayanaamahesh committed:
Level 6: Delete

nayanaamahesh committed:
Added TestCases Draft 2 + Final Clean up for Week 1

nayanaamahesh committed:
Level 7: Save
Implement automatic saving and loading of tasks to/from file

nayanaamahesh committed:
Add Date Feature with relevant Exceptions

nayanaamahesh committed:
New Pull Request to Branch 8

nayanaamahesh committed:
Merge branch 'branch-Level-7' into branch-Level-8

nayanaamahesh committed:
Revert "Merge branch 'branch-Level-7' into branch-Level-8"

This reverts commit fe0c3b734d6ad16b7f47a3b89cd940227fce5b98, reversing
changes made to d294588c41da26f71d7c12efc31fa29708b1028d.

nayanaamahesh committed:
Revert "Revert "Merge branch 'branch-Level-7' into branch-Level-8""

This reverts commit d2121d3c2ebb8ae8e75b395a9fb4901e60de6ed1.

nayanaamahesh committed:
Add to Branch 8 (Forgot to create a branch 8)

nayanaamahesh committed:
Add More OOP

nayanaamahesh committed:
Add More OOP
Add JavaDocs

nayanaamahesh committed:
Organize into Packages

nayanaamahesh committed:
Merge remote-tracking branch 'origin/add-gradle-support'

nayanaamahesh committed:
Add JUnit Tests for Parser and TaskList

nayanaamahesh committed:
Level 9: Find
Update ParserTest.java accordingly

LeeZeHao commented on src/main/java/nayana/Parser.java:
```diff
@@ -0,0 +1,160 @@
+import java.time.format.DateTimeParseException;
+import nayana.command.*;
+import nayana.task.*;
+public class Parser {
```
According to coding standards, perhaps there should be a class JavaDoc comment here.

LeeZeHao commented on src/main/java/nayana/TaskList.java:
```diff
@@ -0,0 +1,130 @@
+package nayana;
+
+import nayana.task.*;
```
I think wildcard imports are not recommended in the coding standard. Could this be changed?

LeeZeHao commented on src/main/java/nayana/Ui.java:
```diff
@@ -0,0 +1,153 @@
+package nayana;
+import java.util.ArrayList;
+import java.util.Scanner;
+import nayana.task.*;
```
Similar wildcard imports are found in multiple other classes, I think they should be replaced to comply with code standards?

LeeZeHao commented on src/main/java/nayana/Parser.java:
```diff
@@ -0,0 +1,160 @@
+import nayana.task.*;
+public class Parser {
+    /**
+     * Parses the user input and returns the corresponding Command object.
```
Public methods have JavaDoc comments from what I can see, this is good!

LeeZeHao reviewed (commented):
Overall follows code standards, looks good!
Only some small documentation / import changes needed

nayanaamahesh committed:
Add CheckStyle

nayanaamahesh committed:
Merge branch 'A-CheckStyle'

* A-CheckStyle:
  Add CheckStyle

samuelthen commented on src/main/java/nayana/Parser.java:
```diff
@@ -0,0 +1,196 @@
+package nayana;
+
+import java.time.LocalDate;
```
I like how the imported classes are imported neatly.

samuelthen commented on src/test/java/nayana/ParserTest.java:
```diff
@@ -0,0 +1,88 @@
+    void testParseValidCommands() throws NayanaException {
+        // Test valid commands
+        Command command = Parser.parse("bye");
+        assertTrue(command instanceof ExitCommand, "Command should be an instance of ExitCommand.");
```
Perhaps you can consider using `assertInstanceOf`?

samuelthen reviewed (commented):
### Overall Comments

Overall, very **well done**! The codebase is well structured!

Just a few suggestions given.

nayanaamahesh committed:
JavaFx Tutorial Part 1,2,3

nayanaamahesh committed:
JavaFX setting up FXML

nayanaamahesh committed:
Add ChatBot Features to GUI

nayanaamahesh committed:
Merge branch 'Level-10'

* Level-10:
  Add ChatBot Features to GUI
  JavaFX setting up FXML
  JavaFx Tutorial Part 1,2,3

nayanaamahesh committed:
Update Parser and FindCommand classes with GUI enhancements
The Parser and FindCommand classes were updated to integrate the new GUI features for displaying tasks and command output. This revision ensures the user interface is synchronized with backend changes and provides a more intuitive experience.

Previously, the app relied on console-based interaction. The transition to a GUI required modifications to methods like printFoundTasks() to support graphical output. Additionally, error handling for invalid commands was refined to improve user feedback through the UI.

These updates improve overall usability, providing a final implementation for week 5.

nayanaamahesh committed:
Add Assertions
A-Assertions c1

nayanaamahesh committed:
Subject:
Refactor methods: avoid long methods

Body:
Long methods reduce code readability and maintainability. They can be difficult to understand, test, and debug. By splitting long methods into smaller, more manageable methods, we improve the clarity and maintainability of the code.

Let's:
Identify and extract lengthy methods into smaller, well-named methods.
Ensure each new method has a single responsibility.
Update method calls and references accordingly.

Refactoring methods in this way enhances code quality and makes the codebase easier to work with in the future. It also supports better unit testing and debugging by isolating functionality into discrete units.

nayanaamahesh committed:
Subject:
Refactor code: improve readability and maintainability
Body:
Code readability and maintainability are crucial for a high-quality codebase. This commit addresses several code quality issues to make the code clearer and more maintainable:

Let's:
Simplify code by avoiding deep nesting and complicated expressions.
Replace magic numbers with named constants to improve clarity.
Refactor code to make the primary functionality (happy path) prominent and obvious.
Practice the KISS (Keep It Simple, Stupid) principle by avoiding over-engineering and unnecessary complexity.
Avoid premature optimizations that may complicate the code without providing tangible benefits.
Reduce multiple levels of abstraction within code fragments to enhance readability.
Ensure that the code is easy to understand and does not 'trip up' the reader by being straightforward and intuitive.

These changes improve the overall quality of the code, making it easier to understand, maintain, and extend. The code now adheres to best practices for readability and simplicity, supporting better future development and debugging.

nayanaamahesh committed:
Subject:
Enhance code: add comments and remove unsafe practices
Body:
Improving code quality involves not only refactoring for readability but also ensuring safety and clarity. This commit addresses these areas:

Let's:
Add meaningful code comments to clarify complex logic and enhance understanding for future developers.
Remove unsafe practices, including potential security vulnerabilities and code that could lead to runtime errors or unexpected behaviors.
Update code to adhere to best practices for safety and robustness.

These changes make the codebase safer and more comprehensible, facilitating easier maintenance and reducing the risk of errors. Clear comments and the elimination of unsafe practices contribute to a higher standard of code quality and developer collaboration.

nayanaamahesh committed:
Merge pull request #2 from nayanaamahesh/A-Assertions

Add Assertions

nayanaamahesh committed:
Merge branch 'A-CodeQuality' into master

nayanaamahesh committed:
Merge branch 'master' of github.com:nayanaamahesh/ip

* 'master' of github.com:nayanaamahesh/ip:
  Subject: Enhance code: add comments and remove unsafe practices Body: Improving code quality involves not only refactoring for readability but also ensuring safety and clarity. This commit addresses these areas:
  Subject: Refactor code: improve readability and maintainability Body: Code readability and maintainability are crucial for a high-quality codebase. This commit addresses several code quality issues to make the code clearer and more maintainable:
  Subject: Refactor methods: avoid long methods
  Add Assertions A-Assertions c1

nayanaamahesh committed:
Update README.md

nayanaamahesh committed:
Merge branch 'master' of github.com:nayanaamahesh/ip

* 'master' of github.com:nayanaamahesh/ip:
  Update README.md

nayanaamahesh committed:
Subject:
Add Reminder feature for upcoming tasks
Body:
Introduced a new Reminder feature that alerts users about tasks due within the upcoming week. This enhancement automatically checks for deadlines and events scheduled in the next seven days, providing a proactive notification to help users stay on track.The feature was added to improve task management and ensure that users don't miss important deadlines. It integrates seamlessly with the existing task list, parsing, and GUI, and is triggered each time the user keys in "remind".
This feature enhances user productivity and supports better planning by keeping upcoming tasks visible.

nayanaamahesh committed:
Subject:
Refactor Remind and Find methods to use Streams
Body:
Refactored the Remind and Find methods to utilize Java Streams for improved readability and performance.

Changes:
Remind Method: Replaced traditional loops with Stream API operations to filter and collect upcoming tasks due within the week. This enhances the clarity and efficiency of the code.
Find Method: Converted the search logic to use Streams for a more concise and expressive approach to task filtering based on search criteria.

Using Streams not only simplifies the code but also leverages modern Java features for better performance and maintainability. This refactoring supports a more functional programming style and aligns with current best practices.

nayanaamahesh committed:
Merge pull request #4 from nayanaamahesh/B-Reminders

Subject:

nayanaamahesh committed:
Merge branch 'master' of github.com:nayanaamahesh/ip

* 'master' of github.com:nayanaamahesh/ip:
  Subject: Add Reminder feature for upcoming tasks Body: Introduced a new Reminder feature that alerts users about tasks due within the upcoming week. This enhancement automatically checks for deadlines and events scheduled in the next seven days, providing a proactive notification to help users stay on track.The feature was added to improve task management and ensure that users don't miss important deadlines. It integrates seamlessly with the existing task list, parsing, and GUI, and is triggered each time the user keys in "remind". This feature enhances user productivity and supports better planning by keeping upcoming tasks visible.

nayanaamahesh committed:
Merge pull request #5 from nayanaamahesh/A-Streams

Merge Streams to Master

nayanaamahesh committed:
Merge branch 'master' of github.com:nayanaamahesh/ip

* 'master' of github.com:nayanaamahesh/ip:
  Subject: Refactor Remind and Find methods to use Streams Body: Refactored the Remind and Find methods to utilize Java Streams for improved readability and performance.

nayanaamahesh committed:
Change Logic for File Path
Error is shown in Window rather than CLI

nayanaamahesh committed:
Add Better GUI
Tweak GUI to match asymmetric nature of the conversation
Highlight Errors (Nayana shows done face)
Allow Resizing
Adjust profile pic size

nayanaamahesh committed:
Updated Style to comply with CheckStyle
Used Ai to generate some JavaDocs (AI-Assisted)
Tests Updated

nayanaamahesh committed:
Add representative screenshot

nayanaamahesh committed:
Update README

nayanaamahesh committed:
Added Better GUI
Added Comment for Better GUI is required as it helps for better user interface.
This is done by changing the background of user to blue.

nayanaamahesh committed:
Removed getResponse Method
Method was used to simple GUI Version of Nayana
No longer used so removed to reduce clutter

nayanaamahesh committed:
Changed Comments and JavaDoc
Nayana is the name of the new application.
So removed all traces of Duke.

nayanaamahesh committed:
Update README.md

nayanaamahesh committed:
Update javadoc for ExitCommand

nayanaamahesh committed:
Update javadoc for ListCommand

nayanaamahesh committed:
Update javadocs for MarkCommand

nayanaamahesh committed:
Update javadoc for UnmarkCommand
